drivers/i2s/i2s_ll_stm32.c

```diff
@@ -837,37 +857,46 @@ struct queue_item rx_1_ring_buf[CONFIG_I2S_STM32_RX_BLOCK_COUNT + 1];
 struct queue_item tx_1_ring_buf[CONFIG_I2S_STM32_TX_BLOCK_COUNT + 1];
 
 static struct i2s_stm32_data i2s_stm32_data_1 = {
-	.dma_name = I2S1_DMA_NAME,
 	.rx = {
-		.dma_channel = I2S1_DMA_CHAN_RX,
+		.dma_name = DT_I2S_1_DMA_CONTROLLER_RX,
+		.dma_channel = DT_I2S_1_DMA_CHANNEL_RX,
 		.dma_cfg = {
 			.block_count = 1,
-			.dma_slot = I2S1_DMA_SLOT_RX,
+			.dma_slot = DT_I2S_1_DMA_SLOT_RX,
 			.channel_direction = PERIPHERAL_TO_MEMORY,
 			.source_data_size = 1,  /* 16bit default */
 			.dest_data_size = 1,    /* 16bit default */
 			.source_burst_length = 0, /* SINGLE transfer */
 			.dest_burst_length = 1,
+			.channel_priority = DT_I2S_1_DMA_PRIORITY_RX,
 			.dma_callback = dma_rx_callback,
 		},
+		.src_addr_increment = DT_I2S_1_DMA_SRC_ADDR_INCREMENT_RX,
```

Comment:
Trying to understand how this works globally.
From https://git.kernel.org/pub/scm/linux/kernel/git/devicetree/devicetree-rebasing.git/plain/Bindings/dma/stm32-dma.txt, I'd expect a mask is applied to get this info from bitmap

Reply:
The bitmap is decoded in the gen_defines.py

Reply:
ok, got it, then we need maintainer point of view on this regard.
Not sure this is the right place.